### ISSUE TITLE
GDScript stack underflow while yielding to signals (regression from #28884)

### ISSUE THREAD

**Issue by Xrayez** [godotengine/godot#30269]
**Godot version:**
3.2-dev 4d99408d123e4eea5773c8d1e35b59385b0aefa1

Regression from #28884.

**OS/device including version:**
Windows 10

**Issue description:**
While running GUT tests encountered this:

```
Debugger Break, Reason: 'Stack Underflow (Engine Bug)'
ERROR: GDScriptLanguage::debug_get_stack_level_function: Index p_level=0 out of size (_debug_call_stack_pos=0)
   At: modules\gdscript\gdscript_editor.cpp:271
ERROR: GDScriptLanguage::debug_get_stack_level_line: Index p_level=0 out of size (_debug_call_stack_pos=0)
   At: modules\gdscript\gdscript_editor.cpp:260
ERROR: GDScriptLanguage::debug_get_stack_level_source: Index p_level=0 out of size (_debug_call_stack_pos=0)
   At: modules\gdscript\gdscript_editor.cpp:280
*Frame 0 - :-1 in function ''
Enter "help" for assistance.
debug>
```

**Steps to reproduce:**
```
git clone https://github.com/bitwes/Gut/
```
Now open the project, run main scene (default). Godot Test Window shall appear.
Select `res://test/unit/test_gut_yielding.gd` from the popup menu [`...`] button:

![2019-07-04](https://user-images.githubusercontent.com/17108460/60627077-9d1d4400-9df5-11e9-9648-0570734cacec.png)

Press `Run`. Stalls on the first yield test (`test_can_yield_using_built_in_timer`).

**Comment by StraToN** [comment]
I am affected by "stack underflow" as well. I have no idea of the cause though.

**Comment by vnen** [comment]
I have a ton of yields in my project and it doesn't happen, hard to tell what's causing it.

Also, you don't need to use `while`, the `completed` signal is only emitted when the function didn't yield anymore.

**Comment by Xrayez** [comment]
@vnen figured the steps to reproduce, see updated original report. 🙂

@StraToN I'm mostly running the tests via command line as part of CI-like tests in my project via pre-commit hook. And yeah I do something very similar to what you described in your snippet.

**Comment by owstetra** [comment]
@vnen i also have the same problem , i also have a lot of yields but for some reason some of them not causing the problem but others do
commenting the yield line will stop causing it

**Comment by volzhs** [comment]
I'm facing it too.

**Comment by magpie514** [comment]
I also seem affected by it. It happens when a yield should happen, indeed. (on current master at the time of writing, d897131ac) 
![screenshot1](https://user-images.githubusercontent.com/323651/60750426-74897b80-9fa8-11e9-8cab-053f4092f395.png)

**Comment by zaksnet** [comment]
![Screenshot_5](https://user-images.githubusercontent.com/1850856/60864487-6ba8dc00-a22c-11e9-8d9f-69eec227bcb9.png)
Same for me.

**Comment by muchitto** [comment]
I got the same "Stack Underflow (Engine Bug)" error message just now. I have multiple levels of `yield(state, "completed")` -kinda code in my game (wait for the yield to complete in the calling function and in that calling function and in that etc up the stack frame...). I just made some changes to my code and added one more "yield completed" signal thing, the engine spouted that error message to me. If I remove that one yield, everything works... 

Are too many nested yield complete things much for the engine or what is happening? :)

I'm using freshly compiled godot from the master branch, that I just pulled like hour or two ago...
